Filestream inputs with duplicated ID cause Filebeat to panic

First things first: having two Filestream inputs configured with the same ID will lead to data duplication, this is not a valid configuration.

- Version: main, 8.9 branch
- Operating System: all

**Steps to Reproduce**
1. Create a `filebeat.yml` with the following content
   ```yaml
   filebeat:
     inputs:
       - type: filestream
         id: foo
         paths:
           - /tmp/foo
       - type: filestream
         id: foo
         paths:
           - /tmp/foo2
   output:
     file:
       enabled: true
   ```
2. Run Filebeat
3. It will panic:
```
panic: name foo already used

goroutine 90 [running]:
github.com/elastic/elastic-agent-libs/monitoring.panicErr(...)
        /home/tiago/go/pkg/mod/github.com/elastic/elastic-agent-libs@v0.3.9/monitoring/registry.go:257
github.com/elastic/elastic-agent-libs/monitoring.(*Registry).Add(0xc0003a9840, {0xc0003a46b8, 0x0}, {0x45f3b60, 0xc00072d380}, 0x30)
        /home/tiago/go/pkg/mod/github.com/elastic/elastic-agent-libs@v0.3.9/monitoring/registry.go:155 +0xda
github.com/elastic/elastic-agent-libs/monitoring.(*Registry).NewRegistry(0xc0003a9840, {0xc0003a46b8, 0x3}, {0x0?, 0x0?, 0x4078f36?})
        /home/tiago/go/pkg/mod/github.com/elastic/elastic-agent-libs@v0.3.9/monitoring/registry.go:94 +0x165
github.com/elastic/beats/v7/libbeat/monitoring/inputmon.NewInputRegistry({0x407584e, 0xa}, {0xc0003a46b8, 0x3}, 0xc00013bb10?)
        /home/tiago/devel/beats/libbeat/monitoring/inputmon/input.go:64 +0x3e6
github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile.NewMetrics({0xc0003a46b8, 0x3})
        /home/tiago/devel/beats/filebeat/input/filestream/internal/input-logfile/metrics.go:66 +0xa5
github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile.(*managedInput).Run(_, {0xc000488c30, {0xc0003a46b8, 0x3}, {{0x40708ec, 0x8}, {0x40708ec, 0x8}, {0x406b23e, 0x6}, ...}, ...}, ...)
        /home/tiago/devel/beats/filebeat/input/filestream/internal/input-logfile/input.go:64 +0x1bd
github.com/elastic/beats/v7/filebeat/input/v2/compat.(*runner).Start.func1()
        /home/tiago/devel/beats/filebeat/input/v2/compat/compat.go:122 +0x215
created by github.com/elastic/beats/v7/filebeat/input/v2/compat.(*runner).Start
        /home/tiago/devel/beats/filebeat/input/v2/compat/compat.go:119 +0xd8
```

The problem comes from registering metrics for Filestream inputs using the ID: https://github.com/elastic/beats/blob/59447f67558e4ea0445211c33735d7877bb32e41/filebeat/input/filestream/internal/input-logfile/metrics.go#L66
when the second input tries to registry its metrics, the monitoring/metrics code panics.

We need to find a way to detect this case and prevent the panic from happening.

--- a/CHANGELOG.next.asciidoc
+++ b/CHANGELOG.next.asciidoc
@@ -142,7 +142,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Fix CEL input JSON marshalling of nested objects. {issue}35763[35763] {pull}35774[35774]
 - Fix metric collection in GCPPubSub input. {pull}35773[35773]
 - Fix end point deregistration in http_endpoint input. {issue}35899[35899] {pull}35903[35903]
-- Fix duplicate ID panic in filestream metrics. {issue}35964[35964] {pull}1[1]
+- Fix duplicate ID panic in filestream metrics. {issue}35964[35964] {pull}35972[35972]
 
 *Heartbeat*
 

--- a/CHANGELOG.next.asciidoc
+++ b/CHANGELOG.next.asciidoc
@@ -142,6 +142,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Fix CEL input JSON marshalling of nested objects. {issue}35763[35763] {pull}35774[35774]
 - Fix metric collection in GCPPubSub input. {pull}35773[35773]
 - Fix end point deregistration in http_endpoint input. {issue}35899[35899] {pull}35903[35903]
+- Fix duplicate ID panic in filestream metrics. {issue}35964[35964] {pull}1[1]
 
 *Heartbeat*
 

--- a/filebeat/input/filestream/internal/input-logfile/input.go
+++ b/filebeat/input/filestream/internal/input-logfile/input.go
@@ -30,6 +30,7 @@ import (
 
 type managedInput struct {
 	userID           string
+	metricsID        string
 	manager          *InputManager
 	ackCH            *updateChan
 	sourceIdentifier *sourceIdentifier
@@ -61,7 +62,7 @@ func (inp *managedInput) Run(
 	defer cancel()
 	ctx.Cancelation = cancelCtx
 
-	metrics := NewMetrics(ctx.ID)
+	metrics := NewMetrics(inp.metricsID)
 	defer metrics.Close()
 
 	hg := &defaultHarvesterGroup{

--- a/filebeat/input/filestream/internal/input-logfile/manager.go
+++ b/filebeat/input/filestream/internal/input-logfile/manager.go
@@ -177,10 +177,13 @@ func (cim *InputManager) Create(config *conf.C) (v2.Input, error) {
 			" duplication, please add an ID and restart Filebeat")
 	}
 
+	metricsID := settings.ID
 	cim.idsMux.Lock()
 	if _, exists := cim.ids[settings.ID]; exists {
 		cim.Logger.Errorf("filestream input with ID '%s' already exists, this "+
-			"will lead to data duplication, please use a different ID", settings.ID)
+			"will lead to data duplication, please use a different ID. Metrics "+
+			"collection has been disabled on this input.", settings.ID)
+		metricsID = ""
 	}
 
 	// TODO: improve how inputs with empty IDs are tracked.
@@ -223,6 +226,7 @@ func (cim *InputManager) Create(config *conf.C) (v2.Input, error) {
 		manager:          cim,
 		ackCH:            cim.ackCH,
 		userID:           settings.ID,
+		metricsID:        metricsID,
 		prospector:       prospector,
 		harvester:        harvester,
 		sourceIdentifier: sourceIdentifier,

